ARCHITECTURE.md

```diff
@@ -0,0 +1,100 @@
+> Each output file will be also available with `compressed` / `minified` content.
+
+![Tools Lifecycle](https://cloud.githubusercontent.com/assets/4987015/17671967/0c55b916-631a-11e6-9d79-d99dd50f630a.png)
+
+### How to run Angular in NodeJS
+
+To be able to run an Angular module in NodeJS the tools need to mock a browser environment.
+
+> Material Tools does only mock the most necessary parts of the browser so the mock is as small as possible.
+
+This created mock will then be assigned to the global `window` variable.
+> The global `window` variable is bound to the NodeJS `globals`, which is shared between all files.
+
+This can lead to situations, where the tools are interfering with the user's build process.
+
+- The mock is polluting the `globals` of NodeJS.
+- The NodeJS [module](https://nodejs.org/api/modules.html) cache will be polluted with wrong exports.
+
+Read more about the way we solved that issue with the [Virtual Context](#under-the-hood-understanding-the-virtual-context)
+
+### Why are some type of files compiled manually
+
+Some developers are expecting Material Tools to export the Angular Material stylesheet without the layouts included by default.
+
+Since the `core` module of Angular Material always includes the `layouts` by default, we are not able to retrieve the `core` module
+without the `layouts` included.
+
+To work around this issue, we have to manually compile the `core` module with explicitly excluding the `layout` stylesheets.
+
+### Under-the-Hood: Understanding the Virtual Context
+Material Tools places a high value on on the isolation of the modified NodeJS environment.
+
+The Virtual Context allows the tools to run specific code completely isolated in 
+another [V8](https://developers.google.com/v8/) context. <br/>
+It is not only isolating the given code. It also provides a minimalistic NodeJS environment.
+
+> It is also possible to require other files from the Virtual Context without leaving the isolated context.<br/>
+
+A minimal `require` method has been created inside of the plain V8 context, which supports:
+- Nesting of the isolated context
+- Creating / Overwriting context global variables (e.g `window`)
+- Enabling [strict mode](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Strict_mode)
+
+It plays such a big role in Material Tools, because it allows us to completely isolate our changes from the original NodeJS context
+and also allows us to overwrite the `globals` as mentioned previously.
+
+The overwriting of the `globals` is important to be able to run an Angular application in NodeJS.
+
+> That's why the Virtual Context is also responsible for mocking the most necessary parts of a browser to run an Angular application.
+
+<img src="https://cloud.githubusercontent.com/assets/4987015/17678625/2e07fa6a-6338-11e6-9fe6-e6ee54dec53e.png" height="210">
+
+### Under-the-Hood: How to generate a static theme
```

Comment:
Move this section up to the top below "Build a Custom Set"
Title as "Build a Custom, Static Theme"